torch/nn/functional.py

```diff
@@ -1515,6 +1515,8 @@ def binary_cross_entropy_with_logits(input, target, weight=None, size_average=Tr
                 observations for each minibatch depending on :attr:`size_average`. When :attr:`reduce`
                 is ``False``, returns a loss per input/target element instead and ignores
                 :attr:`size_average`. Default: ``True``
+        pos_weight (Tensor, optional): a weight of positive examples.
+                Must be a vector with length equal to the number of classes.
```

Comment:
Isn't the number of classes always 2 in this case? The description is pretty confusing

Reply:
Of course, you're right. BCE is binary by definition. I meant multilabel classification.

What about this?

> Must be a vector with length equal to the number of labels where
> each label is corresponding to a binary class.

Reply:
Sorry, I still don't think I understand. This loss wants an input and target of the same shape, but the target should be a binary tensor. Now, how should `pos_weight` look like? Is it just a 2 element tensor?

Reply:
Also, I'm wondering if we really want to add this. You can easily weight samples according to even more complicated schemes if you only use `reduce=False`. In particular, I think this one is equivalent to the code below, which isn't too complicated.
```python
(F.binary_cross_entropy_with_logits(..., reduce=False) * class_weights).mean()
```
Here `class_weights` allows you to weight both things, if you want to get the `pos_weight` behavior then you do
```python
class_weights = torch.ones_like(input).mask_fill_(target == 1, pos_weight_value)
```

Reply:
`class_weights` are the weights of labels. `pos_weight` acts *inside* BCE formula for every particular label.

**The usual BCE**:

`target * -log(sigmoid(input)) + (1 - target) * -log(1 - sigmoid(input))`

For multilabel classification we can use vectors for `target` and `input`.

**The proposed BCE**:

`pos_weight * target * -log(sigmoid(input)) + (1 - target) * -log(1 - sigmoid(input))`

`pos_weight` is multiplied by only the "positive" part. It allows us to "increase" the fraction of positive examples in imbalanced datasets, so we can trade off precision and recall. I don't know any simple way to do it *outside* BCE.

TF already have [a similar function](https://www.tensorflow.org/api_docs/python/tf/nn/weighted_cross_entropy_with_logits).

For example. If we have 7 labels and the batch size is equal to 4. Then:
- input — 4×7
- target — 4×7
- pos_weight — 1×7 (or just 7, thanks to broadcasting) — a vector of weights of positive part for every label.